### DISCUSSION TITLE
There are no consumers for the queue defined through definitions.json file in rabbitmq in kubernetes cluster

### DISCUSSION THREAD

**Discussion by gbalamk** [rabbitmq/rabbitmq-server#10003] [Other] [answered]
### Describe the bug

I have defined the logs queues, exchange and binding in rabbitmq configuration but when it is deployed it didn't create any exchanges and consumers for this queue. And the messages are not delivering and are piling up in the queue in rabbitmq portal.
I have 3 node cluster deployed in kubernetes cluster. 

Earlier it was working fine but suddenly I got this issue. Also I have redployed the application with the same configuration but still I am facing the same issue. I don't see any abnormalities in the logs. Attaching the logs and config details.

Rabbit MQ version : 3.9.14-management
ERLang Version : 24.3.3 
STOMP protocol : AMQP 0-9-1

Below is screenshot of the logs queue in my application. 

![image](https://github.com/rabbitmq/rabbitmq-server/assets/86233219/7987c51a-8738-477d-9bcb-a946e8540357)

Please help me to resolve this issue.

### Reproduction steps

1.Attaching rabbitmq deployment file and dockerfile to reproduce the issue.
2.
[rabbitmq-deploy.txt](https://github.com/rabbitmq/rabbitmq-server/files/13484808/rabbitmq-deploy.txt)
[rabbitmq-dockerfile.txt](https://github.com/rabbitmq/rabbitmq-server/files/13484809/rabbitmq-dockerfile.txt)
[rabbit1.log](https://github.com/rabbitmq/rabbitmq-server/files/13484813/rabbit1.log)
[rabbit2.log](https://github.com/rabbitmq/rabbitmq-server/files/13484815/rabbit2.log)
[rabbit3.log](https://github.com/rabbitmq/rabbitmq-server/files/13484819/rabbit3.log)

3.
...

### Expected behavior

With this configuration, it should create consumers for the logs_unicast queue which I showed in the above screenshot.

**Comment by michaelklishin** [comment, marked as answer]
Your applications add consumers. RabbitMQ (with the exception of shovels and federation links) never adds consumers to queues.

It does not matter if a queue was declared by a client or created via a definitions files: your applications are responsible for registering consumers.

Perhaps make sure that your apps that are supposed to consume those messages are deployed, up and running, and [can connect to RabbitMQ nodes](https://rabbitmq.com/logging.html#logged-events).

**Comment by michaelklishin** [comment]
In other news, RabbitMQ 3.9 has [reached end of life several months ago](https://rabbitmq.com/versions.html).

**Reply by gbalamk** [reply]
Sorry to bother you, But it was working before and we do have another rabbitmq running in docker swarm, even that is configured the same way as the above and it's version is 3.8.5. Below is the screenshot of that rabbitmq application.

![image](https://github.com/rabbitmq/rabbitmq-server/assets/86233219/98920483-9b3d-42a9-bf5a-1ce9ec26232c)

**Reply by michaelklishin** [reply]
@gbalamk 3.8 has reached EOL in mid-2202.

**Your applications** connect to RabbitMQ and register consumers. If you don't have any consumers online, either they are not running or they fail to connect to the cluster.

This is NOT an issue in RabbitMQ if your applications are not online or do not register consumers.

**Comment by michaelklishin** [comment]
To narrow down issues with application connectivity (or startup):

 * [Connection Lifecycle Events](https://rabbitmq.com/logging.html#logged-events) logged by nodes (always start with the logs)
 * [Troubleshooting Network Connectivity](https://rabbitmq.com/troubleshooting-networking.html)
 * [Troubleshooting TLS](https://rabbitmq.com/troubleshooting-ssl.html) (if TLS is used)

From there, collecting and inspecting application logs may provide relevant clues.
